### PR TITLE
feat!: use wags-tails for managing data

### PR DESCRIPTION
**Pull request by jsstevenson**
Use `wags-tails` for acquiring and handling source data files.

Also makes some small changes to console feedback -- would be good to return to holistically.

overall vibes: crossing fingers emoji

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -15,6 +15,8 @@ The [etl,pg] argument tells pip to install packages to fulfill the dependencies
 
 ### External requirements
 
+# TODO update this
```
Comment (jsstevenson) [inline README.md:18]: lol

File: src/disease/etl/omim.py [hunk 2]
```diff
@@ -1,36 +1,46 @@
...
 
 class OMIM(Base):
     """Gather and load data from OMIM."""
 
-    def _extract_data(self, use_existing: bool = False) -> None:
-        """Override parent extract method to enforce OMIM-specific data file
-        requirements.
-        :param use_existing: technically non-functional, but included to match
-        sibling method signatures. If True, will print warning but otherwise proceed.
+    def _omim_access_warning(self) -> None:
```
Comment (jsstevenson) [inline src/disease/etl/omim.py:14]: todo change name of this

File: pyproject.toml [hunk 3]
```diff
@@ -35,8 +35,8 @@ dependencies = [
 dynamic = ["version"]
 
 [project.optional-dependencies]
-pg = ["psycopg[binary]"]
-etl = ["owlready2==0.40", "rdflib", "requests", "typing-extensions", "bioversions"]
+pg = ["psycopg[binary]", "requests"]
+etl = ["owlready2==0.40", "rdflib", "typing-extensions", "wags-tails"]
```
Comment (jsstevenson) [inline pyproject.toml:39]: todo remove typing-extensions (+ in pipfile)

File: src/disease/etl/base.py [hunk 4]
```diff
@@ -1,41 +1,57 @@
...
+        self._data_source: Union[
+            NcitData, OncoTreeData, MondoData, DoData, CustomData
+        ] = self._get_data_handler(data_path)  # type: ignore
         self._database = database
-        self._src_dir: Path = Path(data_path / self._src_name.value.lower())
         self._store_ids = self.__class__.__name__ in SOURCES_FOR_MERGE
         if self._store_ids:
             self._added_ids = []
 
+    def _get_data_handler(self, data_path: Optional[Path]) -> DataSource:
```
Comment (korikuzma) [inline src/disease/etl/base.py:47]:
```suggestion
    def _get_data_handler(self, data_path: Optional[Path] = None) -> DataSource:
```

File: src/disease/etl/omim.py [hunk 5]
```diff
@@ -1,36 +1,46 @@
...
-        sibling method signatures. If True, will print warning but otherwise proceed.
+    def _raise_access_error(self) -> None:
+        """Raise improper data access error and describe proper data access."""
+        raise FileNotFoundError(
+            "Could not locate OMIM data. Per README, OMIM "
+            "source files must be manually placed in "
+            f"{self._data_source.data_dir.absolute().as_uri()}"
+        )
+
+    def _get_data_handler(self, data_path: Optional[Path]) -> DataSource:
```
Comment (korikuzma) [inline src/disease/etl/omim.py:22]:
```suggestion
    def _get_data_handler(self, data_path: Optional[Path] = None) -> DataSource:
```

**Review by korikuzma — APPROVED** [review]
so clean

**Pull request merged**